assets/js/components/dashboard-sharing/ModuleRecoveryAlert.js

```diff
@@ -0,0 +1,230 @@
+import Button from '../Button';
+import Checkbox from '../Checkbox';
+import BannerNotification from '../notifications/BannerNotification';
+import ProgressBar from '../ProgressBar';
+import Spinner from '../Spinner';
+
+const { useDispatch, useSelect } = Data;
+
+export default function ModuleRecoveryAlert() {
+	const [ checkboxes, setCheckboxes ] = useState( null );
+	const [ recoveringModules, setRecoveringModules ] = useState( false );
+
+	const recoverableModules = useSelect( ( select ) =>
+		select( CORE_MODULES ).getRecoverableModules()
+	);
+
+	const userAccessibleModules = useSelect( ( select ) => {
+		const modules = select( CORE_MODULES ).getRecoverableModules();
+
+		if ( modules === undefined ) {
+			return undefined;
+		}
+
+		const accessibleModules = Object.keys( modules ).map( ( slug ) => ( {
+			slug,
+			hasModuleAccess: select( CORE_MODULES ).hasModuleAccess( slug ),
+		} ) );
+
+		if (
+			accessibleModules.some(
+				( { hasModuleAccess } ) => hasModuleAccess === undefined
+			)
+		) {
+			return undefined;
+		}
+
+		return accessibleModules
+			.filter( ( { hasModuleAccess } ) => hasModuleAccess )
+			.map( ( { slug } ) => slug );
+	} );
+
+	const { recoverModule } = useDispatch( CORE_MODULES );
+
+	const updateCheckboxes = useCallback(
+		( slug ) =>
+			setCheckboxes( ( currentCheckboxes ) => ( {
+				...currentCheckboxes,
+				[ slug ]: ! currentCheckboxes[ slug ],
+			} ) ),
+		[]
+	);
+
+	const handleRecoverModules = useCallback( () => {
+		setRecoveringModules( true );
+		const modulesToRecover = Object.keys( checkboxes ).filter(
+			( module ) => checkboxes[ module ]
+		);
+		Promise.all(
+			modulesToRecover.map( ( module ) => recoverModule( module ) )
+		).finally( () => setRecoveringModules( false ) );
+	}, [ checkboxes, recoverModule ] );
+
+	useEffect( () => {
+		if ( userAccessibleModules !== undefined && checkboxes === null ) {
+			const checked = {};
+
+			userAccessibleModules.forEach( ( module ) => {
+				checked[ module ] = true;
+			} );
+
+			setCheckboxes( checked );
+		}
+	}, [ checkboxes, userAccessibleModules ] );
+
+	const recoverableModulesList = Object.keys( recoverableModules );
+
+	if (
+		recoverableModules === undefined ||
+		recoverableModulesList.length === 0
+	) {
+		return null;
+	}
+
+	let description;
+	let children;
+
+	if ( userAccessibleModules === undefined || checkboxes === null ) {
+		description = null;
+		children = <ProgressBar />;
+	} else if ( userAccessibleModules.length === 0 ) {
+		if ( recoverableModulesList.length === 1 ) {
+			description = sprintf(
+				/* translators: %s: module name. */
+				__(
+					'%s data was previously shared with other users on the site by another admin who no longer has access. To restore access, the module must be recovered by another admin who has access.',
+					'google-site-kit'
+				),
+				recoverableModules[ recoverableModulesList[ 0 ] ].name
+			);
+			children = null;
```

Comment:
We can set `null` by default to these variables and don't need to do it in conditions:
```suggestion
	let description = null;
	let children = null;

	if ( userAccessibleModules === undefined || checkboxes === null ) {
		children = <ProgressBar />;
	} else if ( userAccessibleModules.length === 0 ) {
		if ( recoverableModulesList.length === 1 ) {
			description = sprintf(
				/* translators: %s: module name. */
				__(
					'%s data was previously shared with other users on the site by another admin who no longer has access. To restore access, the module must be recovered by another admin who has access.',
					'google-site-kit'
				),
				recoverableModules[ recoverableModulesList[ 0 ] ].name
			);
```